src/m3ninx/search/searcher/conjunction.go

```diff
@@ -49,13 +51,26 @@ func (s *conjunctionSearcher) Search(r index.Reader) (postings.List, error) {
 		pl           postings.List
 		plNeedsClone = true
 	)
+
+	listCount := len(s.searchers)
+	if listCount < len(s.negations) {
+		listCount = len(s.negations)
+	}
+	lists := make([]postings.List, 0, listCount)
+
 	for _, sr := range s.searchers {
 		curr, err := sr.Search(r)
 		if err != nil {
 			return nil, err
 		}
+		lists = append(lists, curr)
+	}
 
-		// TODO: Sort the iterators so that we take the intersection in order of increasing size.
+	sort.Slice(lists, func(i, j int) bool {
```

Comment:
nit: not sure how performance critical this code is, but `sort.Sort()` is generally faster alternative.

Reply:
https://github.com/m3db/m3/pull/3981/commits/e8a43625b4797d4349e4493aa37822aaf16abd53